File: contents/bogo_sort/code/java/bogo.java
public class Bogo {
    static void bogoSort(int[] arr) {
        while(!isSorted(arr)) {
            shuffle(arr);
        }
    }

    static boolean isSorted(int[] arr) {
        for (int i = 0; i < arr.length - 1; i++) {
            if(arr[i] > arr[i + 1]) {
                return false;
            }
        }

        return true;
    }

    static void shuffle(int[] arr) {
        for (int r = arr.length - 1; r > 0; r--) {
            int i = (int) Math.floor(Math.random() * r);
            int tmp = arr[i];
            arr[i] = arr[r];
            arr[r] = tmp;
        }
    }


    public static void main(String[] args) {
        int[] test = new int[]{20, -3, 50, 1, -6, 59};

        System.out.println("Unsorted array :");
        for (int i = 0; i < test.length; i++) {
            System.out.print(test[i] + " ");
        }


        bogoSort(test);


        System.out.println("\n\nSorted array :");
        for (int i = 0; i < test.length; i++) {
            System.out.print(test[i] + " ");
        }
	System.out.println("");
    }
}

File: contents/bogo_sort/code/julia/bogo.jl
using Random

function is_sorted(a::Vector{Float64})
    for i = 1:length(a)-1
        if (a[i] > a[i + 1])
            return false
        end
    end
    return true
end

function bogo_sort!(a::Vector{Float64})
    while(!is_sorted(a))
        shuffle!(a)
    end
end

function main()
    a = [1.0, 3.0, 2.0, 4.0]
    bogo_sort!(a)
    println(a)
end

main()
